### PR TITLE
proxyd: Filter out archive requests

### PR DESCRIPTION
**Pull request by Inphi**
**Description**
Change targets proxyd-3.8.4

### CODE REVIEW

File: go/proxyd/rpc.go [hunk 1]
```diff
@@ -152,3 +164,52 @@ func IsBatch(raw []byte) bool {
...
+	return blockNum+uint64(BLOCK_NUM_ARCHIVE_HEIGHT) < latestBlockNum, nil
+}
+
+// quick hack: if the request looks like a request containing a quantity|tag at the end, then return the tag
+func extractTerminalBlockTag(req *RPCReq) (string, error) {
+	var input []json.RawMessage
+	if err := json.Unmarshal(req.Params, &input); err != nil {
+		return "", err
+	}
+	if len(input) < 2 {
```
Comment (tynes) [inline go/proxyd/rpc.go:209]: are there any requests that take only a block tag?

Comment (Inphi) [reply go/proxyd/rpc.go:209]: also note that that function is only [called for eth_call, eth_getBalance and eth_getTransactionCount rpcs ](https://github.com/ethereum-optimism/optimism/pull/2635/files#diff-326039202cfd8c96893c534b0a1be54828dbaf252544893173db3d3d85579448R177)

File: go/proxyd/rpc.go [hunk 2]
```diff
@@ -152,3 +164,52 @@ func IsBatch(raw []byte) bool {
 	}
 	return false
 }
+
+func isArchiveRequest(req *RPCReq, getBlockNum GetLatestBlockNumFn) (bool, error) {
+	const BLOCK_NUM_ARCHIVE_HEIGHT = 256
```
[the remark this answers is not in the document]
Comment (tynes) [reply go/proxyd/rpc.go]: `geth` does hold on to archival state for the last 256 blocks, trailing the tip

[the remark this answers is not in the document]
Comment (Inphi) [reply go/proxyd/rpc.go]: l2geth?

[the remark this answers is not in the document]
Comment (tynes) [reply go/proxyd/rpc.go]: Its fine to reduce it to be safe

**Comment by tynes** [comment]
The code generally looks good to me, the tests are failing which is not ideal

File: go/proxyd/rpc.go (hunk 1 above)
Comment (tynes) [reply go/proxyd/rpc.go:209]: all of those take more than 2 elements so we good

File: go/proxyd/rpc.go [hunk 3]
```diff
@@ -152,3 +164,52 @@ func IsBatch(raw []byte) bool {
...
+	// NOTE: we're reading from an in-memory getBlockNum LVC, which doesn't block
+	if latestBlockNum, err = getBlockNum(context.Background()); err != nil {
+		return false, err
+	}
+
+	blockNum, err := decodeBlockInput(tag)
+	if err != nil {
+		return false, err
+	}
+	return blockNum+uint64(BLOCK_NUM_ARCHIVE_HEIGHT) < latestBlockNum, nil
```
Comment (tynes) [inline go/proxyd/rpc.go:200]: Should this be `>`?

Comment (Inphi) [reply go/proxyd/rpc.go:200]: hmm no. We return true (i.e. it's an archive request) if `blockNum` is too old/ too far behind.

**Pull request closed without merging**